js/views/modals/purchase/Purchase.js

```diff
@@ -38,10 +42,14 @@ export default class extends BaseModal {
     // add the item to the order.
     this.order.get('items').add(item);
 
+    this.receipt = this.createChild(Receipt, {
```

Comment:
It happens because the events need to be re-delegated when you append a new render, which I forgot about. 

I'm refactoring the action (pay, etc.) button to be a separate view and ended up moving all the sub-views to being created in the render, which should fix this.